xapian-core/tests/api_sorting.cc

```diff
@@ -319,3 +319,49 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
     );
     return true;
 }
+
+DEFINE_TESTCASE(replace_weights, backend) {
+    Xapian::Database mydb(get_database("apitest_onedoc"));
+    Xapian::Enquire enquire(mydb);
+    enquire.set_query(Xapian::Query("word"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    // old_max_possible = 0.269763689697702
+    double old_max_possible = mymset.get_max_possible();
+    double new_weight = 0.25;
+    vector<double> weights {new_weight};
+    mymset.replace_weights(weights.begin(), weights.end());
+    Xapian::MSetIterator i = mymset.begin();
+    TEST_EQUAL_DOUBLE(i.get_weight(), new_weight);
+    TEST_EQUAL_DOUBLE(mymset.get_max_attained(), new_weight);
+    TEST_EQUAL_DOUBLE(mymset.get_max_possible(), old_max_possible);
+    return true;
+}
+
+DEFINE_TESTCASE(replace_weights2, backend) {
+    Xapian::Database mydb(get_database("apitest_onedoc"));
+    Xapian::Enquire enquire(mydb);
+    enquire.set_query(Xapian::Query("word"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    vector<double> weights {1.0, 2.0};
+    TEST_EXCEPTION(Xapian::InvalidArgumentError, mymset.replace_weights(weights.begin(), weights.end()));
+    return true;
+}
+
+DEFINE_TESTCASE(sort_existing_mset_by_relevance, backend) {
+    Xapian::Database db = get_database("apitest_simpledata");
+    Xapian::Enquire enquire(db);
+    enquire.set_query(Xapian::Query("word"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    vector<Xapian::docid> docids {*(mymset.begin()), *(++mymset.begin())};
```

Comment:
You can use `mymset[0]` and `mymset[1]`, which will likely look a little less strangely than this.